src/Console/Diagnostic/SourceCodeIntegrityCommand.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Console\Diagnostic;
+
+use Glpi\Console\AbstractCommand;
+use Glpi\System\Diagnostic\SourceCodeIntegrityChecker;
+use Glpi\Toolbox\VersionParser;
+use Symfony\Component\Console\Formatter\OutputFormatter;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Output\StreamOutput;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+
+class SourceCodeIntegrityCommand extends AbstractCommand
+{
+    protected $requires_db = false;
+
+    protected function configure()
+    {
+        parent::configure();
+
+        $this->setName('diagnostic:source_code_integrity');
+        $this->setDescription(__('Check GLPI source code file integrity. This command runs in validation mode by default unless --baseline is specified.'));
+        $this->addOption(
+            'baseline',
+            'b',
+            InputOption::VALUE_NONE,
+            __('Generate a baseline manifest of the file hashes. Does not check the files against the official manifest.'),
+        );
```

Comment:
IMHO, this should be moved into a different command (e.g. `build:generate_code_baseline`) and make this command hidden (unless in dev environment). Indeed, we should not expose the baseline generation feature to end users.